File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3Site/QuoteContractOption3SiteValidator.vb
Namespace Entity

    Namespace QuoteContractOption3Sites

        Public Class QuoteContractOption3SiteValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oQuoteContractOption3Site As QuoteContractOption3Site, _
            ByVal assets As DataView)

                'make sure that contact object is valid
                If oQuoteContractOption3Site.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oQuoteContractOption3Site.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oQuoteContractOption3Site.QuoteContractSiteReference.Trim.Length = 0 Then
                    Me.AddCriticalMessage("* Site Reference Required")
                End If

                If Not (oQuoteContractOption3Site.StartDate < oQuoteContractOption3Site.EndDate) Then
                    Me.AddCriticalMessage("* End Date must be greater than Start Date")
                End If

                If oQuoteContractOption3Site.FirstVisitDate() < oQuoteContractOption3Site.StartDate Or _
                     oQuoteContractOption3Site.FirstVisitDate > oQuoteContractOption3Site.EndDate Then
                    Me.AddCriticalMessage("* First Visit Date must be between Start Date and End Date")
                End If

                If oQuoteContractOption3Site.VisitFrequencyID = 0 Then
                    Me.AddCriticalMessage("* Visit Frequency Required")
                End If

                If assets.Table.Columns.Count < 16 Then
                    Me.AddCriticalMessage("* Please load and fill in the assets schedule.")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If


            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3Site/QuoteContractOption3SiteValidator.cs
using System.Collections;
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace QuoteContractOption3Sites
    {
        // make sure that contact object is valid
        public class QuoteContractOption3SiteValidator : BaseValidator
        {
            public void Validate(QuoteContractOption3Site oQuoteContractOption3Site, DataView assets)
            {

                // make sure that contact object is valid
                if (oQuoteContractOption3Site.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oQuoteContractOption3Site.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oQuoteContractOption3Site.QuoteContractSiteReference.Trim().Length == 0)
                {
                    AddCriticalMessage("* Site Reference Required");
                }

                if (!(oQuoteContractOption3Site.StartDate < oQuoteContractOption3Site.EndDate))
                {
                    AddCriticalMessage("* End Date must be greater than Start Date");
                }

                if (oQuoteContractOption3Site.FirstVisitDate < oQuoteContractOption3Site.StartDate | oQuoteContractOption3Site.FirstVisitDate > oQuoteContractOption3Site.EndDate)
                {
                    AddCriticalMessage("* First Visit Date must be between Start Date and End Date");
                }

                if (oQuoteContractOption3Site.VisitFrequencyID == 0)
                {
                    AddCriticalMessage("* Visit Frequency Required");
                }

                if (assets.Table.Columns.Count < 16)
                {
                    AddCriticalMessage("* Please load and fill in the assets schedule.");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}